lib/task/import/importDipObjectsTask.class.php

```diff
@@ -0,0 +1,585 @@
+    {
+      throw new sfException('Undo log directory does not exist.');
+    }
+  }
+
+  /**
+   * Return file pointer resource for the first CSV file found within the DIP's
+   * objects directory. CSV file's extension must be lower-case.
+   *
+   * @return resource  File pointer resource for CSV file
+   */
+  protected function openFirstCsvFile()
+  {
+    // Find all CSV files in the objects directory
+    $csvFiles = sfFinder::type('file')->name('*.csv')->in($this->dipDir .'/objects');
+
+    // Attempt to open the first CSV file found
+    if (0 == count($csvFiles) || false === $fh = fopen($csvFiles[0], 'rb'))
+    {
+      throw new sfException('You must specify a DIP that contains a CSV file');
+    }
+
+    return $fh;
+  }
+
+  /**
+   * Determine CSV column order and which information object attribute (slug or
+   * identifier) will be used as unique value to specify information objects to
+   * timport digital objects to. Read CSV data, creating an array of information
+   * object unique values and the original filenames of digital objects that
+   * should be imported and associated with these information objects.
+   *
+   * @param resource $fh  File pointer resource for CSV file
+   * @param string $objectsPath  Path to DIP object files
+   *
+   * @return array  Hash with hash key being unique value and hash value(s)
+   *                being original object filenames
+   */
+  protected function parseCsvData($fh, $objectsPath)
+  {
+    // Create a lookup table to match original file basename (UUID and
+    // extension removed) with original object filename (example:
+    // 979c4458-21f3-11e1-a4bd-001d09282b9d-foo.tif)
+    $filenames = $this->createFilenameLookup($objectsPath);
+
+    // Determine column order and what type of unique value will be used to find
+    // information objects (slug or identifier)
+    $this->processCsvHeaderRow($fh);
+
+    $digitalObjects = array();
+
+    // Build hash on information_object key, with value being an array if information
+    // object has multiple digital objects attached
+    while ($row = fgetcsv($fh, 1000))
+    {
+      $filename = $this->getRowColumnValue('filename', $row);
+      $filepath = rtrim($objectsPath, '/') .'/'. $filename;
+
+      // Check if filename has been changed by Archivematica
+      if (!file_exists($filepath))
+      {
+        $key = null;
+
+        // Substitute original file extension (e.g. TIFF) with new DIP extension
+        // (e.g. JPEG)
+        if (preg_match('/(.+)\.(\w{3})$/', $filename, $matches))
+        {
+          $key = strtolower($matches[1]);
+        }
+
+        if (isset($key) && isset($filenames[$key]))
+        {
+          $filepath = str_replace($filename, $filenames[$key], $filepath);
+        }
+        else
+        {
+          throw new Exception("Error: Couldn't find file $filepath");
+        }
+      }
+
+      $uniqueValue = $this->getRowColumnValue($this->uniqueValueColumnName, $row);
```

Comment:
Same here, could we get the filename 'uniqueValueColumnName' index only once instead of making am array_search() for each row?